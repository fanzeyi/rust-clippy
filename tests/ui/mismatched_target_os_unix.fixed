// run-rustfix

#![warn(clippy::mismatched_target_os)]
#![allow(unused)]

#[cfg(target_os = "linux")]
fn linux() {}

#[cfg(target_os = "freebsd")]
fn freebsd() {}

#[cfg(target_os = "dragonfly")]
fn dragonfly() {}

#[cfg(target_os = "openbsd")]
fn openbsd() {}

#[cfg(target_os = "netbsd")]
fn netbsd() {}

#[cfg(target_os = "macos")]
fn macos() {}

#[cfg(target_os = "ios")]
fn ios() {}

#[cfg(target_os = "android")]
fn android() {}

#[cfg(target_os = "emscripten")]
fn emscripten() {}

#[cfg(target_os = "fuchsia")]
fn fuchsia() {}

#[cfg(target_os = "haiku")]
fn haiku() {}

#[cfg(target_os = "illumos")]
fn illumos() {}

#[cfg(target_os = "l4re")]
fn l4re() {}

#[cfg(target_os = "redox")]
fn redox() {}

#[cfg(target_os = "solaris")]
fn solaris() {}

#[cfg(target_os = "vxworks")]
fn vxworks() {}

// list with conditions
#[cfg(all(not(any(target_os = "solaris", target_os = "linux")), target_os = "freebsd"))]
fn list() {}

// correct use, should be ignored
#[cfg(target_os = "freebsd")]
fn correct() {}

fn macro_use() {
    if cfg!(target_os = "android") {}
    if cfg!(target_os = "dragonfly") {}
    if cfg!(target_os = "emscripten") {}
    if cfg!(target_os = "freebsd") {}
    if cfg!(target_os = "fuchsia") {}
    if cfg!(target_os = "haiku") {}
    if cfg!(target_os = "illumos") {}
    if cfg!(target_os = "ios") {}
    if cfg!(target_os = "l4re") {}
    if cfg!(target_os = "linux") {}
    if cfg!(target_os = "macos") {}
    if cfg!(target_os = "netbsd") {}
    if cfg!(target_os = "openbsd") {}
    if cfg!(target_os = "redox") {}
    if cfg!(target_os = "solaris") {}
    if cfg!(target_os = "vxworks") {}
    if cfg!(target_os = "hermit") {}
    if cfg!(target_os = "none") {}
    if cfg!(target_os = "wasi") {}
    if cfg!(any(target_os = "linux", target_os = "macos")) {}
    if cfg!(all(not(any(target_os = "solaris", target_os = "linux")), target_os = "freebsd")) {}

    // correct use
    if cfg!(target_os = "macos") {}
}

fn main() {}
