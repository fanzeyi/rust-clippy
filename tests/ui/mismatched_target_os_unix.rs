// run-rustfix

#![warn(clippy::mismatched_target_os)]
#![allow(unused)]

#[cfg(linux)]
fn linux() {}

#[cfg(freebsd)]
fn freebsd() {}

#[cfg(dragonfly)]
fn dragonfly() {}

#[cfg(openbsd)]
fn openbsd() {}

#[cfg(netbsd)]
fn netbsd() {}

#[cfg(macos)]
fn macos() {}

#[cfg(ios)]
fn ios() {}

#[cfg(android)]
fn android() {}

#[cfg(emscripten)]
fn emscripten() {}

#[cfg(fuchsia)]
fn fuchsia() {}

#[cfg(haiku)]
fn haiku() {}

#[cfg(illumos)]
fn illumos() {}

#[cfg(l4re)]
fn l4re() {}

#[cfg(redox)]
fn redox() {}

#[cfg(solaris)]
fn solaris() {}

#[cfg(vxworks)]
fn vxworks() {}

// list with conditions
#[cfg(all(not(any(solaris, linux)), freebsd))]
fn list() {}

// correct use, should be ignored
#[cfg(target_os = "freebsd")]
fn correct() {}

fn macro_use() {
    if cfg!(android) {}
    if cfg!(dragonfly) {}
    if cfg!(emscripten) {}
    if cfg!(freebsd) {}
    if cfg!(fuchsia) {}
    if cfg!(haiku) {}
    if cfg!(illumos) {}
    if cfg!(ios) {}
    if cfg!(l4re) {}
    if cfg!(linux) {}
    if cfg!(macos) {}
    if cfg!(netbsd) {}
    if cfg!(openbsd) {}
    if cfg!(redox) {}
    if cfg!(solaris) {}
    if cfg!(vxworks) {}
    if cfg!(hermit) {}
    if cfg!(none) {}
    if cfg!(wasi) {}
    if cfg!(any(linux, macos)) {}
    if cfg!(all(not(any(solaris, linux)), freebsd)) {}

    // correct use
    if cfg!(target_os = "macos") {}
}

fn main() {}
