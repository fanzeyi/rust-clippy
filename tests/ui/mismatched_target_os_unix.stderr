error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:6:1
   |
LL | #[cfg(linux)]
   | ^^^^^^-----^^
   |       |
   |       help: try: `target_os = "linux"`
   |
   = note: `-D clippy::mismatched-target-os` implied by `-D warnings`
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:9:1
   |
LL | #[cfg(freebsd)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "freebsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:12:1
   |
LL | #[cfg(dragonfly)]
   | ^^^^^^---------^^
   |       |
   |       help: try: `target_os = "dragonfly"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:15:1
   |
LL | #[cfg(openbsd)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "openbsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:18:1
   |
LL | #[cfg(netbsd)]
   | ^^^^^^------^^
   |       |
   |       help: try: `target_os = "netbsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:21:1
   |
LL | #[cfg(macos)]
   | ^^^^^^-----^^
   |       |
   |       help: try: `target_os = "macos"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:24:1
   |
LL | #[cfg(ios)]
   | ^^^^^^---^^
   |       |
   |       help: try: `target_os = "ios"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:27:1
   |
LL | #[cfg(android)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "android"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:30:1
   |
LL | #[cfg(emscripten)]
   | ^^^^^^----------^^
   |       |
   |       help: try: `target_os = "emscripten"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:33:1
   |
LL | #[cfg(fuchsia)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "fuchsia"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:36:1
   |
LL | #[cfg(haiku)]
   | ^^^^^^-----^^
   |       |
   |       help: try: `target_os = "haiku"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:39:1
   |
LL | #[cfg(illumos)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "illumos"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:42:1
   |
LL | #[cfg(l4re)]
   | ^^^^^^----^^
   |       |
   |       help: try: `target_os = "l4re"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:45:1
   |
LL | #[cfg(redox)]
   | ^^^^^^-----^^
   |       |
   |       help: try: `target_os = "redox"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:48:1
   |
LL | #[cfg(solaris)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "solaris"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:51:1
   |
LL | #[cfg(vxworks)]
   | ^^^^^^-------^^
   |       |
   |       help: try: `target_os = "vxworks"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:55:1
   |
LL | #[cfg(all(not(any(solaris, linux)), freebsd))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: did you mean `unix`?
help: try
   |
LL | #[cfg(all(not(any(target_os = "solaris", linux)), freebsd))]
   |                   ~~~~~~~~~~~~~~~~~~~~~
help: try
   |
LL | #[cfg(all(not(any(solaris, target_os = "linux")), freebsd))]
   |                            ~~~~~~~~~~~~~~~~~~~
help: try
   |
LL | #[cfg(all(not(any(solaris, linux)), target_os = "freebsd"))]
   |                                     ~~~~~~~~~~~~~~~~~~~~~

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:63:8
   |
LL |     if cfg!(android) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "android"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:64:8
   |
LL |     if cfg!(dragonfly) {}
   |        ^^^^^---------^
   |             |
   |             help: try: `target_os = "dragonfly"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:65:8
   |
LL |     if cfg!(emscripten) {}
   |        ^^^^^----------^
   |             |
   |             help: try: `target_os = "emscripten"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:66:8
   |
LL |     if cfg!(freebsd) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "freebsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:67:8
   |
LL |     if cfg!(fuchsia) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "fuchsia"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:68:8
   |
LL |     if cfg!(haiku) {}
   |        ^^^^^-----^
   |             |
   |             help: try: `target_os = "haiku"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:69:8
   |
LL |     if cfg!(illumos) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "illumos"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:70:8
   |
LL |     if cfg!(ios) {}
   |        ^^^^^---^
   |             |
   |             help: try: `target_os = "ios"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:71:8
   |
LL |     if cfg!(l4re) {}
   |        ^^^^^----^
   |             |
   |             help: try: `target_os = "l4re"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:72:8
   |
LL |     if cfg!(linux) {}
   |        ^^^^^-----^
   |             |
   |             help: try: `target_os = "linux"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:73:8
   |
LL |     if cfg!(macos) {}
   |        ^^^^^-----^
   |             |
   |             help: try: `target_os = "macos"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:74:8
   |
LL |     if cfg!(netbsd) {}
   |        ^^^^^------^
   |             |
   |             help: try: `target_os = "netbsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:75:8
   |
LL |     if cfg!(openbsd) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "openbsd"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:76:8
   |
LL |     if cfg!(redox) {}
   |        ^^^^^-----^
   |             |
   |             help: try: `target_os = "redox"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:77:8
   |
LL |     if cfg!(solaris) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "solaris"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:78:8
   |
LL |     if cfg!(vxworks) {}
   |        ^^^^^-------^
   |             |
   |             help: try: `target_os = "vxworks"`
   |
   = help: did you mean `unix`?

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:79:8
   |
LL |     if cfg!(hermit) {}
   |        ^^^^^------^
   |             |
   |             help: try: `target_os = "hermit"`

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:80:8
   |
LL |     if cfg!(none) {}
   |        ^^^^^----^
   |             |
   |             help: try: `target_os = "none"`

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:81:8
   |
LL |     if cfg!(wasi) {}
   |        ^^^^^----^
   |             |
   |             help: try: `target_os = "wasi"`

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:82:8
   |
LL |     if cfg!(any(linux, macos)) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: did you mean `unix`?
help: try
   |
LL |     if cfg!(any(target_os = "linux", macos)) {}
   |                 ~~~~~~~~~~~~~~~~~~~
help: try
   |
LL |     if cfg!(any(linux, target_os = "macos")) {}
   |                        ~~~~~~~~~~~~~~~~~~~

error: operating system used in target family position
  --> $DIR/mismatched_target_os_unix.rs:83:8
   |
LL |     if cfg!(all(not(any(solaris, linux)), freebsd)) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: did you mean `unix`?
help: try
   |
LL |     if cfg!(all(not(any(target_os = "solaris", linux)), freebsd)) {}
   |                         ~~~~~~~~~~~~~~~~~~~~~
help: try
   |
LL |     if cfg!(all(not(any(solaris, target_os = "linux")), freebsd)) {}
   |                                  ~~~~~~~~~~~~~~~~~~~
help: try
   |
LL |     if cfg!(all(not(any(solaris, linux)), target_os = "freebsd")) {}
   |                                           ~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 38 previous errors

